src/commands/mobile_app/upload.rs

```diff
@@ -265,10 +261,13 @@ fn normalize_directory(path: &Path) -> Result<TempFile> {
         .compression_method(zip::CompressionMethod::Stored)
         .last_modified_time(DateTime::default());
 
-    for (entry_path, relative_path) in entries {
-        debug!("Adding file to zip: {}", relative_path.display());
+    for entry_path in entries {
+        let zip_path = entry_path.strip_prefix(
+            path.parent().ok_or_else(|| anyhow!("Failed to get parent directory"))?
+        )?.to_owned();
+        debug!("Adding file to zip: {}", zip_path.display());
 
-        zip.start_file(relative_path.to_string_lossy(), options)?;
+        zip.start_file(zip_path.to_string_lossy(), options)?;
```

Comment:
How is this `zip_path` different from the `relative_path` we had before?

Reply:
1. No I think you're right that the for loop refactor was not necessary. I've changed it to be this way, apologies for the thrash.
2. Yes it is but I've added a `ok_or_else` just in case